docs/source/cpp/build_system.rst

```diff
@@ -51,7 +51,7 @@ file into an executable linked with the Arrow C++ shared library:
    find_package(Arrow REQUIRED)
 
    add_executable(my_example my_example.cc)
-   target_link_libraries(my_example PRIVATE arrow_shared)
+   target_link_libraries(my_example PRIVATE Arrow::arrow_shared)
```

Comment:
We recommend `Arrow::arrow_shared`. So this change is preferred.
FYI: Users can still use `arrow_shared` with CMake 3.18 or later for backward compatibility.
See also: https://github.com/apache/arrow/blob/master/cpp/src/arrow/ArrowConfig.cmake.in#L110-L121

BTW, we have a Jira issue to update this document: ARROW-17575
If you take over this, I'm very happy!

Here are some notes for the new our CMake packages:
* Users can use `find_package(Arrow)`, `find_package(Parquet)` and so on as usual but can't use `COMPONENTS` like `find_package(Arrow COMPONENTS parquet)`
  * The old our CMake packages just ignore `COMPONENTS`.
* `Arrow::arrow_shared` is recommended than `arrow_shared`
* `CMAKE_PREFIX_PATH` not `PKG_CONFIG_PATH` should be used to find Apache Arrow C++ that isn't installed to the system location such as `/usr`
  * The old our CMake packages also uses `PKG_CONFIG_PATH` (`.pc` can be used to find the `Arrow` CMake package)

Reply:
Thanks @kou I've assigned the ticket to me, it would help me learn some of these changes! I am closing this PR as it will be covered as documentation changes.

Reply:
Thanks!!!
If you have a trouble, please ask me!